src/ui/components/select-dropdown/index.jsx

```diff
@@ -289,7 +294,14 @@ class SelectDropdown extends Component {
 		);
 	}
 
-	toggleDropdown() {
+	onContainerClick( event ) {
+		// don't toggle the dropdown if the user clicks in the search box
+		if ( event.target === ReactDom.findDOMNode( this.refs.dropdownSearchBox ) )
+			return;
```

Comment:
Would it be a good idea to `preventDefault` the event so it isn't captured by any other target?

Reply:
I don't think so – this broke the little `x` button to clear the search. The main thing I'm focused on here is preventing `toggleDropdown()` from being called – it's not really "handling" the click.